packages/mantine/src/components/code-editor/CodeEditor.tsx

```diff
@@ -249,6 +249,17 @@ export const CodeEditor: FunctionComponent<CodeEditorProps> = (props) => {
                     editor.onDidBlurEditorText(async () => {
                         await editor.getAction('editor.action.formatDocument').run();
                     });
```

Comment:
I was thinking about the solution and I think we should maybe go another route. What do you think of formatting only if there are no errors? I was able to achieve it this way:

```suggestion
                    editor.onDidBlurEditorText(async () => {
                        const markers = monaco.editor.getModelMarkers({resource: editor.getModel().uri});
                        const hasErrors = markers.some(
                            (marker) => marker.severity === loader.__getMonacoInstance().MarkerSeverity.Error,
                        );
                        if (!hasErrors) {
                            await editor.getAction('editor.action.formatDocument').run();
                        }
                    });
```

Reply:
Yeah I like this solution.
I'll quickly test it to see if there are no other unintended behaviors, then I'll switch implementation.

Reply:
I feel like it will be a better UX, assuming the same thing can happen in other langages. Let's say you're editing a file of 200 lines and have an error at line 10, you don't want your cursor to jump at the end of the file.

I think prettier does the same thing and only format when your code is valid

Reply:
I've put the condition to rely on `hasMonacoError` instead, since those are the language syntax errors and any text can still be formatted properly with non-syntax errors.